pwiz_tools/Skyline/SettingsUI/Irt/CalibrateIrtDlg.cs

```diff
@@ -924,4 +942,76 @@ public class StandardPeptide : MeasuredPeptide
     {
         public double Irt { get; set; }
     }
+
+    public class ScoredPeptide
+    {
+        public MeasuredPeptide Peptide { get; }
+        public float Score { get; }
+
+        public ScoredPeptide(MeasuredPeptide peptide, float score)
+        {
+            Peptide = peptide;
+            Score = score;
+        }
+    }
+
+    public class PeptideBucket
+    {
+        private readonly double _maxRetentionTime;
+        private readonly List<ScoredPeptide> _peptides;
+
+        public bool Empty => _peptides.Count == 0;
+
+        private PeptideBucket(double maxRetentionTime)
+        {
+            _maxRetentionTime = maxRetentionTime;
+            _peptides = new List<ScoredPeptide>();
+        }
+
+        public float? Peek()
+        {
+            return !Empty ? (float?) _peptides.First().Score : null;
+        }
+
+        public MeasuredPeptide Pop()
+        {
+            if (Empty)
+                return null;
+            var pep = _peptides.First().Peptide;
+            _peptides.RemoveAt(0);
+            return pep;
+        }
+
+        public static PeptideBucket[] BucketPeptides(IEnumerable<ScoredPeptide> peptides, double[] rtBoundaries)
+        {
+            var buckets = rtBoundaries.OrderBy(x => x).Select(boundary => new PeptideBucket(boundary)).ToArray();
+            var curBucketIdx = 0;
+            var curBucket = buckets[0];
+            foreach (var pep in peptides.OrderBy(pep => pep.Peptide.RetentionTime))

```

Comment:
The sorts seem potentially slow. Probably not a huge issue, but because you essentially use multiples of 1/8 your range, you could also just us an array of 8 indexes and RT math to decide which bucket a specific RT goes into.

bucketIndex = bucketIndexes[(rt - minRT)/8]

bucketIndexes = new[] {0, 1, 1, 2, 2, 3, 3, 4};

Reply:
I realized that the sort isn't necessary, since we already had to do the sort before to find the min/max RT.